packages/material-ui-lab/src/Breadcrumbs/Breadcrumbs.js

```diff
@@ -0,0 +1,194 @@
+    paddingLeft: 2,
+    paddingRight: 2,
+  },
+});
+
+function DefaultSeparator(props) {
+  const { className, separatorText } = props;
+  return (
+    <div className={className}>{separatorText}</div>
+  );
+}
+
+DefaultSeparator.propTypes = {
+  className: PropTypes.object.isRequired,
+  separatorText: PropTypes.string.isRequired,
+};
+
+class Breadcrumbs extends React.PureComponent {
+  state = {
+    expanded: false,
+  };
+
+  getSeparator(props) {
+    const { separator: Separator } = this.props;
+    if (Separator) {
+      let className = props.className;
+      if (typeof Separator === 'function') {
+        return <Separator {...props} className={className} />;
+      }
+      if (React.isValidElement(Separator)) {
+        className = classNames(className, Separator.props.className);
+        return React.cloneElement(Separator, {
+          ...props,
+          className,
+        });
+      }
+    }
+    return <DefaultSeparator {...props} />;
+  }
+
+  expand(e) {
+    e.preventDefault();
+    this.setState({ expanded: true });
+  }
+
+  insertSeparators(items) {
+    const { classes, separatorText, separator } = this.props;
+
+    return items.reduce((arr, v, i, source) => {
+      return i < source.length - 1
+        ? arr.concat(
+          v,
+          this.getSeparator(
+            separator
+              ? {
+                key: `separator-${i}`,
+                className: classes.separator,
+              }
+              : {
+                key: `separator-${i}`,
+                className: classes.separator,
+                separatorText,
+              },
+          ),
+        )
+        : arr.concat(v);
+    }, []);
+  }
+
+  renderItemsBeforeAndAfter() {
+    const { itemsBeforeCollapse, itemsAfterCollapse } = this.props;
+    const allItems = this.renderAllItems();
+    // This defends against someone passing weird data, to ensure that if all
+    // items would be shown anyway, we just show all items without the EllipsisItem
+    if (itemsBeforeCollapse + itemsAfterCollapse >= allItems.length) {
+      return allItems;
+    }
+
+    const beforeItems = allItems.slice(0, itemsBeforeCollapse);
+    const afterItems = allItems.slice(
+      allItems.length - itemsAfterCollapse,
+      allItems.length,
+    );
+
+    return [
+      ...beforeItems,
+      <BreadcrumbCollapsed key="ellipsis" onClick={e => this.expand(e)} />,
+      ...afterItems,
+    ];
+  }
+
+  renderAllItems() {
+    const allNonEmptyItems = Children.toArray(this.props.children);
+    return allNonEmptyItems.map(child => React.cloneElement(child, {}));
+  }
+
+  render() {
+    const {
+      classes,
+      children,
+      maxItems,
+      itemsBeforeCollapse,
+      itemsAfterCollapse,
+      separator,
+      separatorText,
+      ...rest
+    } = this.props;
+
+    if (!children) return <div className={classes.root} {...rest} />;
+
+    return (
+      <div className={classes.root} {...rest}>
+        {this.state.expanded ||
```

Comment:
I understand what it does but for consistency I would prefer `!this.state.expanded && ...`. It's easier to recognize since it's the [recommended approach to conditional rendering](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator). I was honestly not sure if react only skips rendering of `false` or any boolean.